colossalai/fx/passes/topo.py

```diff
@@ -0,0 +1,153 @@
+from typing import Dict, List
+from dataclasses import dataclass
+
+# This class is a middleware between partition splitter
+# and Pipeline Scheduler. It records the graph info about
+# partition input/output and provides it to scheduler.
+@dataclass
+class ValPosition:
```

Comment:
I think topology related classes, such as Topo, Partition, should be moved into colossalai/pipeline, because it is not only used in  auto partitioning, but for customized partitioning as well. You can create a middleware folder if you want.

Reply:
How about colossalai/pipeline/middleware

Reply:
It looks fine to me.

Reply:
Done